### PR TITLE
Complex array literal expressions

### PR DESCRIPTION
**Pull request by Schaeff**
addresses #114 

- concat `A + B`
- repeat `A*`

Todo:
- [x] Remove need for parentheses (parser)
- [x] Naming

### CODE REVIEW

File: src/parser/powdr.lalrpop [hunk 1]
```diff
@@ -329,6 +329,12 @@ PublicReference: String = {
     ":" <Identifier>
 }
 
+ArrayExpression: ArrayExpression = {
```
[the remark this answers is not in the document]
Comment (Schaeff) [reply src/parser/powdr.lalrpop:332]: Is `[1] + [0]` still a literal?

[the remark this answers is not in the document]
Comment (chriseth) [reply src/parser/powdr.lalrpop:332]: No, but an ArrayLiteralExpression ;)

[the remark this answers is not in the document]
Comment (chriseth) [reply src/parser/powdr.lalrpop:332]: Oh and could you move it to the pil part? I think it is now in the asm part.

**Review by leonardoalt — APPROVED** [review]
should eventually add tests for the `None` case

**Comment by Schaeff** [comment]
> should eventually add tests for the `None` case

done
